src/gerbv.c

```diff
@@ -623,6 +868,17 @@ gerbv_open_image(gerbv_project_t *gerbvProject, gchar const* filename, int idx,
 	parsed_image = NULL;
     }
     
+    
+    if (parsed_image) {
+        // Assign image (file) attributes if applicable
+        if (attrbuf[0])
```

Comment:
==2924607== Conditional jump or move depends on uninitialised value(s)
==2924607==    at 0x486B96C: gerbv_open_image (gerbv.c:878)
==2924607==    by 0x486C477: gerbv_open_layer_from_filename_with_color (gerbv.c:253)
==2924607==    by 0x11903D: main (main.c:948)

Reply:
Solved by doing this:

```diff
diff --git a/src/gerbv.c b/src/gerbv.c
index 37d4793..ab6b7b5 100644
--- a/src/gerbv.c
+++ b/src/gerbv.c
@@ -588,6 +588,7 @@ gerbv_open_image(gerbv_project_t *gerbvProject, gchar const* filename, int idx,
     // layer.  It also indicates how many layers to expect.  E.g. if --layers=1,2,3,4 then we know it's a 4-layer
     // board and the bottom layer is #4.
     
+    attrbuf[0] = 0;
     if (layertype == GERBV_LAYERTYPE_RS274X) {
         /* Get *all* layers defined, so can map 'b' to max (bottom) layer.  't' is always 1.
            Copper layer number is always first char: digit(2) 1..9 or t or b.  (Others are not copper).
@@ -685,7 +686,6 @@ gerbv_open_image(gerbv_project_t *gerbvProject, gchar const* filename, int idx,
         g_free(ff);
         // Now that we know what 'b' layer means, go ahead and parse this instance.
         ff = x2attr_get_field_or_default(instance, file_functions, "");
-        attrbuf[0] = 0;
         ffp = ff;
         switch (tolower(*ffp++)) {
         case '1': case '2': case '3': case '4': case '5': case '6': case '7': case '8': case '9': 
```